From df891b5ab996744891f5334c27c33be2cbb6a696 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:39:00 -0400
Subject: [x86] cpufeature: Define the Intel MPX feature flag

Message-id: <1406040016-3289-146-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86224
O-Subject: [RHEL7 PATCH v2 145/221] x86, cpufeature: Define the Intel MPX feature flag
Bugzilla: 1076194 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Qiaowei Ren <qiaowei.ren@intel.com>

Define the Intel MPX (Memory Protection Extensions) CPU feature flag
in the cpufeature list.

Signed-off-by: Qiaowei Ren <qiaowei.ren@intel.com>
Link: http://lkml.kernel.org/r/1386375658-2191-2-git-send-email-qiaowei.ren@intel.com
Signed-off-by: Xudong Hao <xudong.hao@intel.com>
Signed-off-by: Liu Jinsong <jinsong.liu@intel.com>
Signed-off-by: H. Peter Anvin <hpa@linux.intel.com>
(cherry picked from commit 191f57c137bcce0e3e9313acb77b2f114d15afbb)
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/include/asm/cpufeature.h b/arch/x86/include/asm/cpufeature.h
index bae697f..8035e0c 100644
--- a/arch/x86/include/asm/cpufeature.h
+++ b/arch/x86/include/asm/cpufeature.h
@@ -216,6 +216,7 @@
 #define X86_FEATURE_ERMS (9*32+ 9) /* Enhanced REP MOVSB/STOSB */
 #define X86_FEATURE_INVPCID (9*32+10) /* Invalidate Processor Context ID */
 #define X86_FEATURE_RTM  (9*32+11) /* Restricted Transactional Memory */
+#define X86_FEATURE_MPX  (9*32+14) /* Memory Protection Extension */
 #define X86_FEATURE_RDSEED (9*32+18) /* The RDSEED instruction */
 #define X86_FEATURE_ADX  (9*32+19) /* The ADCX and ADOX instructions */
 #define X86_FEATURE_SMAP (9*32+20) /* Supervisor Mode Access Prevention */
-- 
1.7.1